### PR TITLE
Deprecate TBB.

### PR DESCRIPTION
**Pull request by joaander**
## Description

Deprecate the use of TBB. Warn the user when `num_cpu_threads > 1` and an operation would use more than one thread.

## Motivation and context

See  #1650.

## How has this been tested?

I verified that the `pytest` scripts generated the new warnings in the expected configurations.

## Change log

```
Deprecated:

* ``num_cpu_threads > 1``. Use ``num_cpu_threads = 1``.
```

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/CONTRIBUTING.rst).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

**Review by janbridley — APPROVED** [review]
Looks good to me

File: hoomd/hpmc/pytest/depletants_sphere.py
Comment (tommy-waltmann) [inline hoomd/hpmc/pytest/depletants_sphere.py:1]: Does this file have something to do with TBB deprecation?

**Review by tommy-waltmann — APPROVED** [review]
Looks good, just one quick question.

**Comment by tommy-waltmann** [comment]
As I grep the codebase, I see that there are other parts of hoomd not modified in this PR (for example, IntegratorHPMCMono, the cluster updater) that utilize tbb code. In the PR, there is only one place that a deprecation warning has been added, should more warnings be added to those classes as well?

File: hoomd/hpmc/pytest/depletants_sphere.py
Comment (joaander) [reply hoomd/hpmc/pytest/depletants_sphere.py:1]: It is an unused file I discovered while testing the changed code. It is somewhat related to TBB as depletants is one of the code paths that uses TBB.

**Comment by joaander** [comment]
> As I grep the codebase, I see that there are other parts of hoomd not modified in this PR (for example, IntegratorHPMCMono, the cluster updater) that utilize tbb code. In the PR, there is only one place that a deprecation warning has been added, should more warnings be added to those classes as well?

@tommy-waltmann, this PR adds warnings to 
* The Python class `HPMCIntegrator` when depletants are enabled (which covers the use of TBB in `IntegratorHPMCMono`)
* The Python class `CPPPotentialUnion` (which covers the use of TBB in PatchEnergyJITUnion.cc).

I did not add a warning to `Clusters`. That code path exists, but is no longer used as it requires TBB < 2021:
https://github.com/glotzerlab/hoomd-blue/blob/935c9b17adfefaec1ea97375ab7f3bad9920c35e/hoomd/hpmc/UpdaterClusters.h#L30-L32
https://github.com/glotzerlab/hoomd-blue/blob/935c9b17adfefaec1ea97375ab7f3bad9920c35e/hoomd/hpmc/UpdaterClusters.h#L368-L372

**Pull request merged**